File: 34. Find First and Last Position of Element in Sorted Array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int first = -1;
        int last = -1;
        if(nums.size() < 1){
            return {-1,-1};
        }
        vector<int> ans;
        for(int i=0;i<nums.size();i++){
            if(nums[i] == target){
                first = i;
                break;
            }
        }
        for(int i=nums.size()-1;i>=0;i--){
            if(nums[i] == target){
                last = i ;
                break;
            }
        }
        ans.push_back(first);
        ans.push_back(last);
        return ans;   
    }
};

File: 34. Find First and Last Position of Element in Sorted Array.java
class Solution {
   public int binarySearch(int[] arr, int target, boolean FirstIndex) {
		int ans = -1;
		int start = 0, end = arr.length - 1;
		while (start <= end) {
			int mid = start + (end - start) / 2;
			if (arr[mid] == target) {
				ans = mid;
				if (FirstIndex) {
					end = mid - 1;
				} else {
					start = mid + 1;
				}
			} else if (arr[mid] > target) {
				end = mid - 1;
			} else {
				start = mid + 1;
			}
		}
		return ans;
	}

	public int[] searchRange(int[] nums, int target) {
		int[] ans = { -1, -1 };
		ans[0] = binarySearch(nums,target,true);
		ans[1] = binarySearch(nums,target,false);
        // System.gc();
		return ans;
	}
}
